backends/platform/sdl/sdl-window.cpp

```diff
@@ -164,6 +168,19 @@ void SdlWindow::grabMouse(bool grab) {
 #endif
 }
 
+void SdlWindow::setMouseRect(const Common::Rect &rect) {
+	grabRect.x = rect.left;
+	grabRect.y = rect.top;
+	grabRect.w = rect.width();
+	grabRect.h = rect.height();
+
+#if SDL_VERSION_ATLEAST(2, 0, 17)
```

Comment:
The documentation regarding this function is stating that it is available since 2.0.18

```c
/**
 * Confines the cursor to the specified area of a window.
 *
 * Note that this does NOT grab the cursor, it only defines the area a cursor
 * is restricted to when the window has mouse focus.
 *
 * \param window The window that will be associated with the barrier.
 * \param rect A rectangle area in window-relative coordinates. If NULL the
 *             barrier for the specified window will be destroyed.
 * \returns 0 on success or a negative error code on failure; call
 *          SDL_GetError() for more information.
 *
 * \since This function is available since SDL 2.0.18.
 *
 * \sa SDL_GetWindowMouseRect
 * \sa SDL_SetWindowMouseGrab
 */
extern DECLSPEC int SDLCALL SDL_SetWindowMouseRect(SDL_Window * window, const SDL_Rect * rect);
```

Reply:
I don't know what should be done here.
There is no 2.0.17 release (it seems libsdl only release even numbers).
At the time of `SDL_SetWindowMouseRect` addition, SDL version was 2.0.17: https://github.com/libsdl-org/SDL/blob/4b42c05ba1eaaaa9a4ef803acea8f13402271039/include/SDL_version.h

Reply:
Sounds to me like the check should be for 2.0.18 since that's what the documentation says. To the extent that 2.0.17 existed, the function presumably wasn't available there for the entire time?

Reply:
Right, it is more logical and I fixed it.